src/ex2.ml

```diff
@@ -0,0 +1,35 @@
+let loadXMLDoc () =
+  let xmlhttp = XmlHttpRequest.create () in
+  let onreadystatechange () =
+    match xmlhttp##readyState, xmlhttp##status with
+    | XmlHttpRequest.DONE, 200 ->
+      let xmlDoc = xmlhttp##responseXML in
+      let buf = Buffer.create 16 in
+      begin match Js.Opt.to_option xmlDoc with
+        | None -> ()
+        | Some xmlDoc ->
+          let elts = xmlDoc##getElementsByTagName (Js.string "ARTIST") in
+          for i = 0 to elts##length - 1 do
+            match Js.Opt.to_option (elts##item (i)) with
+            | None -> ()
+            | Some elt ->
+              begin match Js.Opt.to_option elt##firstChild with
+                | None -> ()
+                | Some elt ->
+                  begin match Js.Opt.to_option elt##nodeValue with
+                    | None -> ()
+                    | Some elt ->
+                      Buffer.add_string buf (Js.to_string elt);
+                      Buffer.add_string buf "<br>"
```

Comment:
Wouldn't it be nice to have 
Js.Opt.case_default: 'a Js.Opt.t -> 'b -> ('a -> 'b) -> 'b ?

Also, the order of arguments does not seem optimal to me.
I'd rather have 
Js.Opt.case: (unit -> 'b) -> ('a -> 'b) -> 'a Js.Opt.t -> 'b

I was confused yesterday because I had written something like
Js.Opt.case x##firstChild (fun () -> ...) (fun x -> ...)
and had a strange syntax error due to camlp4 extension.
The fix was to put parenthesis around x##firstChild.